src/selectedtests/work_items/process_work_items.py

```diff
@@ -52,19 +53,32 @@ def process_queued_task_mapping_work_items(
     """
     _clear_in_progress_work(mongo.task_mappings_queue())
     try:
-        while True:
-            out_of_work = _process_one_task_mapping_work_item(
-                evg_api, mongo, after_date, before_date
-            )
-            if out_of_work:
-                break
+        for work_item in _generate_task_mapping_work_items(mongo):
```

Comment:
I started to change this file before I realized it wasn't the bottleneck. It uses a generator to get the work to do, which I think is a little cleaner than looping and then needing to break out of the loop.

Reply:
Makes sense, I also think this code is clearer.

My only worry is that there is almost-identical logic in this file as well for test_mappings, will that be refactored when you add concurrency to test mapping code? Or should I create a separate ticket to refactor that logic similarly?

Reply:
Updated.